src/m365/spo/commands/page/Page.ts

```diff
@@ -11,75 +11,54 @@ export const supportedPageLayouts = ['Article', 'Home', 'SingleWebPartAppPage',
-        });
-    });
+  public static async getPage(name: string, webUrl: string, logger: Logger, debug: boolean, verbose: boolean): Promise<ClientSidePage> {
+    if (verbose) {
+      logger.logToStderr(`Retrieving information about the page...`);
+    }
+
+    const pageName: string = this.getPageNameWithExtension(name);
+
+    const requestOptions: any = {
+      url: `${webUrl}/_api/web/getfilebyserverrelativeurl('${urlUtil.getServerRelativeSiteUrl(webUrl)}/SitePages/${formatting.encodeQueryParameter(pageName)}')?$expand=ListItemAllFields/ClientSideApplicationId`,
+      headers: {
+        'content-type': 'application/json;charset=utf-8',
+        accept: 'application/json;odata=nometadata'
+      },
+      responseType: 'json'
+    };
+
+    const res: PageItem = await request.get<PageItem>(requestOptions);
+    if (res.ListItemAllFields.ClientSideApplicationId !== 'b6917cb1-93a0-4b97-a84d-7cf49975d4ec') {
+      throw `Page ${name} is not a modern page.`;
+    }
+
+    return ClientSidePage.fromHtml(res.ListItemAllFields.CanvasContent1);
   }
 
-  public static checkout(name: string, webUrl: string, logger: Logger, debug: boolean, verbose: boolean): Promise<ClientSidePageProperties> {
-    return new Promise<ClientSidePageProperties>((resolve: (page: ClientSidePageProperties) => void, reject: (error: any) => void): void => {
-      if (verbose) {
-        logger.log(`Checking out ${name} page...`);
-      }
-
-      const pageName: string = this.getPageNameWithExtension(name);
-      const requestOptions: any = {
-        url: `${webUrl}/_api/sitepages/pages/GetByUrl('sitepages/${formatting.encodeQueryParameter(pageName)}')/checkoutpage`,
-        headers: {
-          'accept': 'application/json;odata=nometadata'
-        },
-        responseType: 'json'
-      };
-
-      request
-        .post<ClientSidePageProperties>(requestOptions)
-        .then((pageData: ClientSidePageProperties) => {
-          if (!pageData) {
-            reject(`Page ${name} information not retrieved with the checkout`);
-            return;
-          }
-
-          if (verbose) {
-            logger.log(`Page ${name} is now checked out`);
-          }
-
-          resolve(pageData);
-        }, (error: any): void => {
-          reject(error);
-        });
-    });
+  public static async checkout(name: string, webUrl: string, logger: Logger, debug: boolean, verbose: boolean): Promise<ClientSidePageProperties> {
+    if (verbose) {
+      logger.log(`Checking out ${name} page...`);
+    }
+
+    const pageName: string = this.getPageNameWithExtension(name);
+    const requestOptions: any = {
+      url: `${webUrl}/_api/sitepages/pages/GetByUrl('sitepages/${formatting.encodeQueryParameter(pageName)}')/checkoutpage`,
+      headers: {
+        'accept': 'application/json;odata=nometadata'
+      },
+      responseType: 'json'
+    };
+
+    const pageData: ClientSidePageProperties = await request.post<ClientSidePageProperties>(requestOptions);
```

Comment:
Also here I think we are not bulletproof and we should do `try... catch` with `handleRejectedODataJsonPromise`